lib/sshkey.rb

```diff
@@ -88,6 +89,27 @@ def sha1_fingerprint(key)
       end
     end
 
+    # Convert an existing SSH public key to SSH2 (RFC4716) public key
+    #
+    # ==== Parameters
+    # * ssh_public_key<~String> - "ssh-rsa AAAAB3NzaC1yc2EA...."
+    # * headers<~Hash> - The Key will be used as the header-tag and the value as the header-value
+    #
+    def ssh_public_key_to_ssh2_public_key(ssh_public_key, headers = nil)
+      raise PublicKeyError, "invalid ssh public key" unless SSHKey.valid_ssh_public_key?(ssh_public_key)
+
+      source_format, source_key, source_comment = parse_ssh_public_key(ssh_public_key)
+
+      # Add a 'Comment' Header Field unless others are explicitly passed in
+      headers = {'Comment' => source_comment} if headers.nil? && !source_comment.empty?
+      header_fields = build_ssh2_headers(headers)
+
```

Comment:
There's a bug here (although it manifests itself in `build_ssh2_headers` #L157) that raises a `NoMethodError: undefined method 'empty?' for nil:NilClass` exception when the original `header` param remains nil AND  `source_comment` is empty.

I used pry in sshkey.rb Note: I slightly changed the variable names for testing.
```
def ssh_public_key_to_ssh2_public_key(ssh_public_key, _headers = nil)
[..]
header_fields = build_ssh2_headers(headers || _headers)
```
```
 def build_ssh2_headers(headers = {})
     binding.pry if headers.nil?
```
Then `rake`, and when I was dropped to a pry shell, I jumped `up`.
```
[1] pry(SSHKey)> _headers
=> nil
[2] pry(SSHKey)> source_comment
=> ""
[3] pry(SSHKey)> _headers.nil? && !source_comment.empty?
=> false
```

Here's the additional tests I ran, they should probably be added into the test suite. The second test against `public_key2` without a header caught the bug.
```
+  def test_ssh2_public_key_bits
+    public_key1 = "ssh-rsa #{SSH_PUBLIC_KEY1} me@example.com"
+    public_key2 = "ssh-rsa #{SSH_PUBLIC_KEY2}"
+    public_key3 = "ssh-rsa #{SSH_PUBLIC_KEY3} 1024-bit DSA with provided comment"
+
+    assert_equal(SSH2_PUBLIC_KEY1, SSHKey.ssh_public_key_to_ssh2_public_key(public_key1))
+    assert_equal(SSH2_PUBLIC_KEY2, SSHKey.ssh_public_key_to_ssh2_public_key(public_key2))
+    assert_equal(SSH2_PUBLIC_KEY2, SSHKey.ssh_public_key_to_ssh2_public_key(public_key2), "x-private-use-header: no comment" )
+    assert_equal(SSH2_PUBLIC_KEY3, SSHKey.ssh_public_key_to_ssh2_public_key(public_key3, "x-private-use-header: some value that is long enough to go to wrap around to a new line."))
+  end
```